### PR TITLE
[Feature] Support efficientnet in mmdetection.

### PR DESCRIPTION
**Pull request by jbwang1997**
## Motivation

Continue on PR #6464
Support EfficientNet-b3 backbone in mmdetection.
Add EfficientNet-b3 + FPN model.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+_base_ = [
+    '../_base_/models/retinanet_r50_fpn.py',
+    '../_base_/datasets/coco_detection.py', '../_base_/default_runtime.py'
+]
+
+cudnn_benchmark = True
```
Comment (ZwwWayne) [inline configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py:6]: is this setting necessary?

File: configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    train_cfg=dict(assigner=dict(neg_iou_thr=0.5)))
+# dataset settings
+img_norm_cfg = dict(
+    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
+train_pipeline = [
+    dict(type='LoadImageFromFile'),
+    dict(type='LoadAnnotations', with_bbox=True),
+    dict(
+        type='Resize',
+        img_scale=(896, 896),
```
Comment (ZwwWayne) [inline configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py:40]: we may write a img_size above line 35, then assign img_size in lines 40/43/54

File: mmdet/models/backbones/efficientnet.py [hunk 3]
```diff
@@ -0,0 +1,409 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
```
Comment (ZwwWayne) [inline mmdet/models/backbones/efficientnet.py:2]: Is this file still necessary? Should we use the efficientnet backbone in mmcls?

File: configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py (hunk 1 above)
Comment (jbwang1997) [reply configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py:6]: I directly copy the setting of retinanet_r50_fpn_crop640_50e_coco.py in nas_fpn.

File: mmdet/models/backbones/efficientnet.py (hunk 3 above)
Comment (jbwang1997) [reply mmdet/models/backbones/efficientnet.py:2]: It needs to change several places to use mmcls efficientnet backbone in mmdet.

Comment (ZwwWayne) [reply mmdet/models/backbones/efficientnet.py:2]: If it is necessary, we should add comments to indicates the difference and explain the necessity.

File: configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py (hunk 1 above)
Comment (ZwwWayne) [reply configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py:6]: OK, we can keep that for now. In the next step we should verify the effectiveness of this option and leave a comment to explain if this key is necessary, otherwise we can delete them.

File: mmdet/models/backbones/efficientnet.py (hunk 3 above)
Comment (jbwang1997) [reply mmdet/models/backbones/efficientnet.py:2]: Done.

File: configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py (hunk 2 above)
Comment (jbwang1997) [reply configs/efficientnet/retinanet_effb3_fpn_crop896_8x4_1x_coco.py]: Done.

File: mmdet/models/backbones/efficientnet.py (hunk 3 above)
Comment (jbwang1997) [reply mmdet/models/backbones/efficientnet.py:2]: Done.

**Review by ZwwWayne — APPROVED** [review]
